NitroxServer/Communication/Packets/Processors/WorldSending/AskBatchCellProcessor.cs

```diff
@@ -0,0 +1,68 @@
+﻿using NitroxModel.DataStructures;
+using NitroxModel.Packets;
+using NitroxModel.Packets.WorldSending;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace NitroxServer.Communication.Packets.Processors.WorldSending
+{
+    public class AskBatchCellProcessor : AuthenticatedPacketProcessor<AskBatchCell>
+    {
+        private TcpServer tcpServer;
+        private Dictionary<string, Queue<Chunk>> batchQueues = new Dictionary<string, Queue<Chunk>>();
+
+        public static readonly string folderName = "DefaultSave"; //todo read from config
+
+        public AskBatchCellProcessor(TcpServer tcpServer)
+        {
+            this.tcpServer = tcpServer;
+        }
+
+        public override void Process(AskBatchCell packet, Player player)
+        {
+            if (packet.FromChunkQueue)
+            {
+                ProcessFromQueue(player);
+            }
+            else
+            {
+                ProcessFromChunk(packet, player);
+            }
+        }
+
+        public void ProcessFromQueue(Player player)
+        {
+            lock (player.chunkQueues)
+            {
+                Queue<Chunk> batchQueue = player.chunkQueues;
+                if (batchQueue.Count == 0)
+                {
+                    foreach (string file in Directory.GetFiles(folderName + "\\"))
+                    {
+                        string[] chunkName = Path.GetFileNameWithoutExtension(file).Split('-');
+                        Int3 chunkCoords = new Int3(int.Parse(chunkName[2]), int.Parse(chunkName[3]), int.Parse(chunkName[4]));
+                        byte[] data = File.ReadAllBytes(file);
```

Comment:
Maybe these chunks should be kept in a persistent memory store so they don't have to be loaded from a file every time people request them.  If you did this you can store an ID in the chunk queue(instead of duplicating all of this data in memory) and do a lookup before sending the data.